File: SITA/sitaa/app/src/main/java/com/example/sitaa/model/Seminar.kt
package com.example.sitaa.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Seminar(
    val id: Int,
    @SerializedName("thesis_id")
    val thesisId: Int,
    @SerializedName("student_id")
    val studentId: Int,
    val title: String,
    @SerializedName("seminar_date")
    val seminarDate: Date?,
    val status: String,
    @SerializedName("rejection_reason")
    val rejectionReason: String?,
    @SerializedName("suggested_date")
    val suggestedDate: Date?,
    @SerializedName("created_at")
    val createdAt: Date,
    @SerializedName("updated_at")
    val updatedAt: Date,
    val student: Student
) {
    data class Student(
        val id: Int,
        val name: String,
        val nim: String,
        @SerializedName("profile_photo")
        val profilePhoto: String?
    )

    enum class Status {
        @SerializedName("pending")
        PENDING,
        @SerializedName("approved")
        APPROVED,
        @SerializedName("rejected")
        REJECTED
    }
}

File: SITA/SITA2/models/Seminar.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Seminar = sequelize.define('Seminar', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  thesis_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'thesis_submissions',
      key: 'id'
    }
  },
  student_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  title: {
    type: DataTypes.STRING(255),
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  research_object: {
    type: DataTypes.TEXT,
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  methodology: {
    type: DataTypes.TEXT,
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  seminar_date: {
    type: DataTypes.DATE,
    allowNull: true
  },
  status: {
    type: DataTypes.ENUM('pending', 'approved', 'rejected'),
    defaultValue: 'pending'
  },
  rejection_reason: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  suggested_date: {
    type: DataTypes.DATE,
    allowNull: true
  }
}, {
  tableName: 'seminar_submissions',
  timestamps: true,
  underscored: true
});

module.exports = Seminar;